### PR TITLE
Add support for multinic

### PR DESCRIPTION
**Pull request by adjackura**

### CODE REVIEW

File: agent/Common/Properties/VersionInfo.cs [hunk 1]
```diff
@@ -41,5 +41,5 @@
 // You can specify all the values or you can default the Build and Revision
 // Numbers by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("3.3.0.0")]
-[assembly: AssemblyFileVersion("3.3.0.0")]
+[assembly: AssemblyVersion("3.4.0.0")]
+[assembly: AssemblyFileVersion("3.4.0.0")]
```
Comment (illfelder) [inline agent/Common/Properties/VersionInfo.cs]: Please move this into a separate pull request.

File: agent/GCEWindowsAgent/AddressReader.cs [hunk 2]
```diff
@@ -46,26 +48,25 @@ private List<IPAddress> ParseForwardIpsResult(string[] forwardedIps)
...
 
-        public bool CompareMetadata(List<IPAddress> oldMetadata, List<IPAddress> newMetadata)
+        public bool CompareMetadata(Dictionary<PhysicalAddress, List<IPAddress>> oldMetadata, Dictionary<PhysicalAddress, List<IPAddress>> newMetadata)
         {
             if (oldMetadata == null || newMetadata == null)
             {
                 return oldMetadata == null && newMetadata == null;
             }
-            return new HashSet<IPAddress>(oldMetadata).SetEquals(newMetadata);
+            return oldMetadata.Count == newMetadata.Count && !oldMetadata.Except(newMetadata).Any(); ;
```
Comment (illfelder) [inline agent/GCEWindowsAgent/AddressReader.cs]: Extra semi-colon. How does this compile?

File: agent/GCEWindowsAgent/AddressReader.cs [hunk 3]
```diff
@@ -46,26 +48,25 @@ private List<IPAddress> ParseForwardIpsResult(string[] forwardedIps)
...
             {
-                return null;
+                PhysicalAddress mac = PhysicalAddress.Parse(entry.MAC);
+                data.Add(mac, ParseForwardIpsResult(entry.ForwardedIps));
             }
+            return data;
         }
 
-        public bool CompareMetadata(List<IPAddress> oldMetadata, List<IPAddress> newMetadata)
+        public bool CompareMetadata(Dictionary<PhysicalAddress, List<IPAddress>> oldMetadata, Dictionary<PhysicalAddress, List<IPAddress>> newMetadata)
```
Comment (illfelder) [inline agent/GCEWindowsAgent/AddressReader.cs]: If I recall correctly, there is a 100 character limit (or 110?). Either way, this is exceeding it.

File: agent/GCEWindowsAgent/AddressWriter.cs [hunk 4]
```diff
@@ -24,32 +24,32 @@
...
-                    && networkInterface.Supports(NetworkInterfaceComponent.IPv4))
+                    && networkInterface.Supports(NetworkInterfaceComponent.IPv4)
+                    && (networkInterface.GetPhysicalAddress() == mac))
                 {
                     return networkInterface.GetIPProperties().GetIPv4Properties().Index;
                 }
             }
 
-            throw new InvalidOperationException("Unable to find primary network interface.");
+            throw new InvalidOperationException("Unable to find network interface with address of " + mac.ToString());
```
Comment (illfelder) [inline agent/GCEWindowsAgent/AddressWriter.cs]: The line length here (and maybe below) as well.

Comment (adjackura) [inline agent/GCEWindowsAgent/AddressWriter.cs]: This ones just under

**Comment by illfelder** [comment]
LGTM. Make sure to squash commits.
